scripts/prepare.js

```diff
@@ -1,18 +1,51 @@
 /* eslint-disable no-console */
-/* tslint:disable:no-console */
 const path = require('path');
 const shell = require('shelljs');
 const chalk = require('chalk');
-const fs = require('fs');
-const log = require('npmlog');
+const fs = require('fs-extra');
+
+const { hashElement } = require('folder-hash');
+
 const { babelify } = require('./compile-babel');
 const { tscfy } = require('./compile-tsc');
 
-function getPackageJson() {
+async function getInfo() {
+  const modulePath = path.resolve('./');
+  const packagePath = path.join(modulePath, 'package.json');
+
+  return {
+    modulePath,
+    packagePath,
+    package: await fs.readJSON(packagePath),
+  };
+}
+
+async function createChecksum() {
   const modulePath = path.resolve('./');
 
-  // eslint-disable-next-line global-require,import/no-dynamic-require
-  return require(path.join(modulePath, 'package.json'));
+  const { hash } = await hashElement(path.join(modulePath, 'src'), {
+    folders: { exclude: ['tests', '__mocks__', '__tests__'] },
+    files: {
+      include: ['*.js', '*.jsx', '*.ts', '*.jsx', '*.json'],
+      exclude: ['*.test.*', '*.stories.*'],
+    },
+  });
+
+  return hash;
+}
+
+async function writeChecksum(checksum, info) {
+  const { packagePath, package } = info;
+
+  return fs.writeJSON(packagePath, { ...package, storybook: { checksum } }, { spaces: 2 });
+}
+
+async function compareChecksum(info) {
+  const { checksum: existing } = info.package.storybook || {};
+
+  const future = await createChecksum();
+
+  return existing === future ? existing : future;
```

Comment:
Will this equality check work with comparing objects? Won't you need to use something like `isDeepStrictEqual`?

Reply:
it's not comparing objects it's comparing strings AFAIK.

Reply:
I assumed it was an object because of this line `const { checksum: existing } = info.package.storybook || {};`

Reply:
info.package.storybook.checksum is a string, I call this string "existing"